bundler/spec/commands/update_spec.rb

```diff
@@ -1474,6 +1477,26 @@
         expect(the_bundle).to include_gems "foo 1.5.0", "bar 2.1.1", "qux 1.1.0"
       end
     end
+
+    context "pre" do
+      it "patch preferred" do
+        bundle "update --patch foo bar --pre"
+
+        expect(the_bundle).to include_gems "foo 1.4.5", "bar 2.1.2.pre", "qux 1.0.0"
+      end
+
+      it "minor preferred" do
+        bundle "update --minor foo bar --pre"
+
+        expect(the_bundle).to include_gems "foo 1.5.1", "bar 3.1.0.pre", "qux 1.0.0"
+      end
+
+      it "major preferred" do
+        bundle "update --major foo bar --pre"
+
+        expect(the_bundle).to include_gems "foo 2.0.0.pre", "bar 4.0.0.pre", "qux 1.0.0"
+      end
```

Comment:
I would also add a spec for `--pre` alone, which should work like `--major`, correct? 

How does `update bar --pre` currently work? Does it unlock `foo` to `2.0.0.pre`, so that it can also upgrade `bar` to `4.0.0.pre`? I _think_ that would be the best behavior?

Reply:
Yup, exactly - it works just like `--major`, I think due to https://github.com/rubygems/rubygems/blob/83ba6cff27bd98b79d6a9194f3fd34c2ec87d6f7/bundler/lib/bundler/gem_version_promoter.rb#L30